tests/lib/rule-tester/rule-tester.js

```diff
@@ -5338,4 +5338,623 @@ describe("RuleTester", () => {
+							{
+								assertionOptions,
+								valid: [],
+								invalid: [
+									{
+										code: "eval(foo)",
+										errors: 1,
+									},
+								],
+							},
+						);
+					}, "Invalid cases must have 'errors' value as an array");
+				});
+
+				it("should fail if message array is used", () => {
+					assert.throws(() => {
+						ruleTester.run(
+							"no-eval",
+							require("../../fixtures/testers/rule-tester/no-eval"),
+							{
+								assertionOptions,
+								valid: [],
+								invalid: [
+									{
+										code: "eval(foo)",
+										errors: ["eval sucks."],
+									},
+								],
+							},
+						);
+					}, "Invalid cases must have 'errors' value as an array of objects e.g. { message: 'error message' }");
+				});
+
+				it("should fail if message and messageId are missing", () => {
+					assert.throws(() => {
+						ruleTester.run(
+							"no-eval",
+							require("../../fixtures/testers/rule-tester/no-eval"),
+							{
+								assertionOptions,
+								valid: [],
+								invalid: [
+									{
+										code: "eval(foo)",
+										errors: [
+											{
+												line: 1,
+											},
+										],
+									},
+								],
+							},
+						);
+					}, "Test error must specify either a 'messageId' or 'message'.");
+				});
+
+				it("should fail if message is present", () => {
+					assert.throws(() => {
+						ruleTester.run(
+							"no-eval",
+							require("../../fixtures/testers/rule-tester/no-eval"),
+							{
+								assertionOptions,
+								valid: [],
+								invalid: [
+									{
+										code: "eval(foo)",
+										errors: [
+											{
+												message: "eval sucks.",
+											},
+										],
+									},
+								],
+							},
+						);
+					}, "The test run requires 'messageId' but the error (also) uses 'message'.");
+				});
+
+				it("should pass if messageId is present", () => {
+					ruleTester.run(
+						"no-eval",
+						require("../../fixtures/testers/rule-tester/no-eval"),
+						{
+							assertionOptions,
+							valid: [],
+							invalid: [
+								{
+									code: "eval(foo)",
+									errors: [
+										{
+											messageId: "evalSucks",
+										},
+									],
+								},
+							],
+						},
+					);
+				});
+			});
+		});
+
+		describe("requireLocation", () => {
+			describe("unset", () => {
+				it("should allow shorthand", () => {
+					ruleTester.run(
+						"no-eval",
+						require("../../fixtures/testers/rule-tester/no-eval"),
+						{
+							assertionOptions: {},
+							valid: [],
+							invalid: [
+								{
+									code: "eval(foo)",
+									errors: 1,
+								},
+							],
+						},
+					);
+				});
+			});
+
+			describe("false", () => {
+				it("should allow shorthand", () => {
+					ruleTester.run(
+						"no-eval",
+						require("../../fixtures/testers/rule-tester/no-eval"),
+						{
+							assertionOptions: { requireLocation: false },
+							valid: [],
+							invalid: 1,
```

Comment:
```suggestion
							invalid: [
								{
									code: "eval(foo)",
									errors: 1,
								},
							],
```

Same as https://github.com/eslint/eslint/pull/20247#discussion_r2463952555.